### PR TITLE
Update count_number_of_one_bits.py

### PR DESCRIPTION
**Pull request by JatinR05**
removed the modulo operator as it is very time consuming in comparison to the and operator

### Describe your change:

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Please add a [timeit](https://docs.python.org/3/library/timeit.html#timeit.timeit) (or similar) benchmark to prove that the proposed code is faster than the original code.
There are lots of examples in this repo.  https://github.com/TheAlgorithms/Python/search?q=timeit

File: bit_manipulation/count_number_of_one_bits.py [hunk 1]
```diff
@@ -1,19 +1,51 @@
...
+    >>> get_set_bits_count_using_brian_kernighans_algorithm(0)
     0
-    >>> get_set_bits_count(256)
+    >>> get_set_bits_count_using_brian_kernighans_algorithm(256)
     1
-    >>> get_set_bits_count(-1)
+    >>> get_set_bits_count_using_brian_kernighans_algorithm(-1)
+    Traceback (most recent call last):
+        ...
+    ValueError: the value of input must be positive
```
Comment (cclauss) [inline bit_manipulation/count_number_of_one_bits.py:22]: Zero is not positive.
```suggestion
    ValueError: the value of input must not be negative
```

**Comment by JatinR05** [comment]
I have the code as suggested

**Comment by JatinR05** [comment]
I have updated the code as suggested.

**Comment by cclauss** [comment]
Please run this on your own machine.  The doctest fails.

**Comment by JatinR05** [comment]
I am extremely sorry for the silly mistakes. Please could you check it now

**Comment by cclauss** [comment]
Would it be more self-documenting to change the variable `number` to `bit_length` throughout the script?

Let's make `benchmark()` more readable with something like this.
```
def benchmark() -> None:
    """
    Benchmark code for comparing 2 functions, with different length int values.
    """
    def do_benchmark(number: int) -> None:
        print(f"Benchmark when {number = }:")
        print(f"{get_set_bits_count_using_modulo_operator(number) = }")
        timing = timeit(
            "z.get_set_bits_count_using_modulo_operator(25)",
            setup="import __main__ as z",
        )
        print(f"timeit() runs in {timing} seconds")
        print(f"{get_set_bits_count_using_brian_kernighans_algorithm(number) = }")
        timing = timeit(
            "z.get_set_bits_count_using_brian_kernighans_algorithm(25)",
            setup="import __main__ as z",
        )
        print(f"timeit() runs in {timing} seconds")

    for number in (25, 37, 58, 0):
        do_benchmark(number)
        print()
```

**Comment by JatinR05** [comment]
I guess the variable name "number" is more appropriate as it's representing the number which the user will be providing as the input and not the bit_length.
Regarding the benchmark updates, it's really a great idea. I'll surely update it.

**Comment by JatinR05** [comment]
Thank you.

**Pull request merged**